content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,6 +8,401 @@ preview: true
+  author    User    @relation(fields: [authorId], references: [id], onDelete: NoAction, onUpdate: NoAction)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+### SetNull
+
+- `onDelete: SetNull` The scalar field of the referenced object will be set to `NULL`.
+
+- `onUpdate: SetNull` When updating the identifier of a referenced object, the scalar fields of the referencing objects will be set to `NULL`.
+
+If one or more of the relations scalar fields are required a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User?   @relation(fields: [authorId], references: [id], onDelete: SetNull, onUpdate: SetNull)
+  authorId  Int?
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+##### Result of using `SetNull`?
+
+When deleting a `User`, the `authorId` will be set to `NULL` for all its authored posts. 
+
+When changing a `User`'s `id`, the `authorId` will be set to `NULL` for all its authored posts. 
+
+### SetDefault
+
+- `onDelete: SetDefault` The scalar field of the referencing object will be set to the fields default value.
+
+- `onUpdate: SetDefault` The scalar field of the referencing object will be set to the fields default value.
+
+These require setting a default for the relation scalar field with [`@default`](../../../../reference/api-reference/prisma-schema-reference#default). If no defaults are provided for any of the scalar fields, a runtime error will be thrown.
+
+```prisma file=schema.prisma highlight=4,5;add
+model Post {
+  id          Int     @id @default(autoincrement())
+  title       String
+  authorName  String  @default("anonymous")
+  author      User?   @relation(fields: [authorId], references: [id], onDelete: SetDefault, onUpdate: SetDefault)
+  authorId    Int?
+}
+
+model User {
+  username  String  @id
+  posts     Post[]
+}
+```
+
+##### Result of using `SetDefault`?
+
+When deleting a `User`, its existing authored posts will now be authored by a `User` with `id` "anonymous". 
+
+When the `username` of a `User` changes, its existing authored posts will now be authored by `User` with `username` "anonymous".
+
+## Upgrade paths
+
+There are a couple of paths you can take when upgrading which will give different results depending on the desired outcome.
+
+The following assumes you have upgraded to 2.26.0 and [enabled the preview feature flag](#enable-the-referentialactions-preview-feature).
+
+### Using Introspection
+
+If you [Introspect](../introspection) your database **before modifying your data model**, the referential actions configured at the database level will be reflected in your Prisma Schema. If you have been using Prisma Migrate or `prisma db push` to manage the database schema, these are likely to be the [<=2.25.0 default values](#the-2250-default-referential-actions). 
```

Comment:
I would like to see some explanation of why I would choose this workflow. For instance, is it because introspecting _first_ lets me clearly see in the schema what my defaults are before I make changes and `migrate dev` to change my database? Or is this for introspection-only people, who would then need to manually update their database if they see an `onDelete: Cascade` that they do not like?  The 'before modifying your data model' suggests I've missed a prerequisite step - I would say "if you *start* by introspecting your database, ... {this happens}

For instance:

```
If you start by [introspect](../introspection) your database , the referential actions configured at the database level will be reflected in your Prisma schema. If you have been using Prisma Migrate or `prisma db push` to manage the database schema, these are likely to be the [<=2.25.0 default values](#the-2250-default-referential-actions).

When you introspect, Prisma compares all the foreign keys in the database with the schema. If the SQL statements `ON DELETE` and `ON UPDATE` do **not** match the default values, they will be explicitly set in the schema file.

The most important clause to review after introspection is `onDelete`, which defaults to `Cascade` in 2.25.0 and earlier.

{insert admonition here}

{Draft =>} Make sure you are happy with every case of `onDelete: Cascade` in your schema - if not, modify your schema and push/migrate, or modify the underlying database if you are using an introspection-only workflow.

{Your example here - 'for example, the following schema means that deleting x also deletes y}
```

Reply:
Oh that's a great suggestion! I also felt like something was missing here but didn't know how to put it into words